packages/core/src/components/fields/SchemaField.js

```diff
@@ -243,7 +243,7 @@ function SchemaFieldRender(props) {
   let idSchema = props.idSchema;
   const schema = retrieveSchema(props.schema, rootSchema, formData);
   idSchema = mergeObjects(
-    toIdSchema(schema, null, rootSchema, formData, idPrefix),
+    toIdSchema(schema, idSchema.$id, rootSchema, formData, idPrefix),
```

Comment:
I'm a bit confused as to what exactly this change does. Do you mind explaining a bit as to how this change in the logic fixes the issue? @QingqiShi 

Depending on how complex it is, it might be good for us to add a comment so future users of this code can more easily understand, as well!

Reply:
Gladly.

However I'm not sure if we need much explaining here. The call signature for `toIdSchema` is `toIdSchema(schema, id, rootSchema, formData, idPrefix)`, so I believe passing `null` to the id argument is incorrect to begin with.

More specifically, in the `ArrayField` each item gets an `itemIdSchema` with the id `idSchema.$id + "_" + index`, this index then gets passed to the item through`SchemaField`. The logic after that is a bit fuzzy to me, but I imagine if we don't use it during the generation of new ids, it defaults to 'root'  prefix and this wrong id eventually get passed to the `MultiSchema` component. 

🙏 Thanks again for spending the time to maintain this package.

Reply:
I see, thanks for the explanation! I'm still a bit confused as to how this change would have only affected the ids in arrays in MultiSchemaField. I would have expected that such a change to SchemaField.js would have had a significant effect on _all_ / many more ids generated through idSchema. Do you know why that's not the case?